### PR TITLE
Add compile time warning for C++14 upgrade

### PR DESCRIPTION
**Pull request by cherylEnkidu**
***
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: app/src/include/firebase/cpp_warning.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
```
Comment (jonsimantov) [inline app/src/include/firebase/cpp_warning.h:1]: Could we rename this to cpp_version_warning.h? Just so it's totally clear what it does. (Or even cpp14_warning.h would also be good.)

File: app/src/include/firebase/cpp_warning.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if __cplusplus < 201400
+#warning \
+    "The next major release of the Firebase C++ SDK will drop support for \
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply app/src/include/firebase/cpp_warning.h:19]: Can you fix this by using two separate strings that are concatenated?

"The next major release of the Firebase C++ SDK will drop support for " \
"C++11, setting the new minimum C++ version to C++14."

**Review by jonsimantov — APPROVED** [review]
What about using #pragma to disable the warning as I suggested, rather than switching it to #message which probably developers won't see?

**Comment by cherylEnkidu** [comment]
> What about using #pragma to disable the warning as I suggested, rather than switching it to #message which probably developers won't see?

I tried, but I only find out how to disable warning or turn down error as warning :(

**Review by jonsimantov — DISMISSED** [review]

**Review by a-maurice — APPROVED** [review]

**Pull request merged**